### PR TITLE
Update errno to match asm-generic

### PR DESCRIPTION
**Pull request by nmn35175**
## Summary

Update NuttX error codes to match the values in asm-generic/errno-base.h and asm-generic/errno.h that are used in Linux kernel. This make it easier to port some SDKs that follow the asm-generic error code standard to NuttX.

## Impact

None

## Testing

None

### CODE REVIEW

**Comment by pkarashchenko** [comment]
What is the motivation for this PR? Why should we align with Linux and not Open/Free BSD for example?
I will need some more time to look into these changes.

**Comment by xiaoxiang781216** [comment]
> What is the motivation for this PR? Why should we align with Linux and not Open/Free BSD for example? I will need some more time to look into these changes.

Here is my thought: It's good to align the error to one of open source POSIX OS. Which one is better? Since many complex SoC run Linux/Android on the big core, and RTOS on small core(e.g. sensorhub, audio dsp, modem). aligning to Linux could save many boring conversion code.

**Comment by pkarashchenko** [comment]
Makes sense to me

File: include/errno.h [hunk 1]
```diff
@@ -51,6 +51,8 @@
  * returned by strerror().
  */
 
+/* usr/include/asm-generic/errno-base.h */
```
Comment (pkarashchenko) [inline include/errno.h:54]:
```suggestion
/* linux/include/asm-generic/errno-base.h */
```
just for more clear reference

Comment (nmn35175) [reply include/errno.h:54]: Can it be /* Sync with linux/include/asm-generic/errno-base.h */
or /* Based on linux/include/asm-generic/errno-base.h */ ?

Comment (xiaoxiang781216) [reply include/errno.h:54]: sure

**Comment by xiaoxiang781216** [comment]
@nmn35175 please squash two patch into one.

**Pull request merged**